extensions/smallrye-opentracing/runtime/src/main/java/io/quarkus/smallrye/opentracing/runtime/QuarkusSmallRyeTracingStandaloneVertxDynamicFeature.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.smallrye.opentracing.runtime;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.enterprise.inject.spi.CDI;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+import io.opentracing.Span;
+import io.opentracing.contrib.jaxrs2.internal.SpanWrapper;
+import io.opentracing.tag.Tags;
+import io.quarkus.vertx.http.runtime.CurrentVertxRequest;
+import io.vertx.ext.web.RoutingContext;
+
+@Provider
+public class QuarkusSmallRyeTracingStandaloneVertxDynamicFeature implements DynamicFeature {
+
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
+        context.register(StandaloneFilter.class);
+    }
+
+    public static class StandaloneFilter implements ContainerRequestFilter {
```

Comment:
This request filter won't catch the issues to do with matching for example. Adding `@PreMatching` would fix that, in which the dynamic feature is not needed. But `ContainerResponseFilter` (without DynamicFeature) should be a safer bet in case some earlier priority pre-matching request filters throw an exception

Reply:
@stuartwdouglas looks like your solution is the best one at the moment, registering a listener from `ContainerRequestFilter` given the feedback from @pavolloffay and @objectiser at #5324 

Only please try to make it a Prematching one and it should do

Reply:
@stuartwdouglas Hi Stuart, as I mentioned, if JAX-RS layer throws 404 or something else before the filters on the matched method will run (such as those registered in your PR), then these exceptions won't be traced. Unless Resteasy implements it somehow differently

Reply:
When I changed it to @PreMatch everything failed, I think because the existing tracing filter is not pre match.

Reply:
@stuartwdouglas Hmm... All the filter in your PR does it registers a Vertx request completion handler, but this filter will only be invoked if RestEasy has found a match and if it has not then it won't be registered. So I'm not sure why would moving this handler registration to the earlier/pre-match chain would cause the failure - please note a pre-matching filter should be registered without a dynamic feature which assumes one attaches the filters to some specific classes - may be it failed because of that.
It is not a big deal, we can follow up with a minor issue to tweak things to make sure JAX-RS exceptions thrown from the pre-matching filters are also traced. If there is some time, then please try to register a pre-match filter without a feature, otherwise please merge :-)